src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,180 @@
+        }
+
+        private static byte[] GetUrlHash(string url)
+        {
+            return Crypto.Hash160(Utility.StrictUTF8.GetBytes(url));
+        }
+
+        internal override void Initialize(ApplicationEngine engine)
+        {
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_NodeList), new StorageItem(new NodeList()));
+            engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_RequestId), new StorageItem(BitConverter.GetBytes(0ul)));
+        }
+
+        protected override void PostPersist(ApplicationEngine engine)
+        {
+            base.PostPersist(engine);
+            (UInt160 Account, BigInteger GAS)[] nodes = GetOracleNodes(engine.Snapshot).Select(p => (Contract.CreateSignatureRedeemScript(p).ToScriptHash(), BigInteger.Zero)).ToArray();
+            foreach (Transaction tx in engine.Snapshot.PersistingBlock.Transactions)
+            {
+                //Filter the response transactions
+                OracleResponse response = tx.GetAttribute<OracleResponse>();
+                if (response is null) continue;
+
+                //Remove the request from storage
+                StorageKey key = CreateStorageKey(Prefix_Request).Add(response.Id);
+                OracleRequest request = engine.Snapshot.Storages[key].GetInteroperable<OracleRequest>();
+                engine.Snapshot.Storages.Delete(key);
+
+                //Remove the id from IdList
+                key = CreateStorageKey(Prefix_IdList).Add(GetUrlHash(request.Url));
+                IdList list = engine.Snapshot.Storages.GetAndChange(key).GetInteroperable<IdList>();
+                if (!list.Remove(response.Id)) throw new InvalidOperationException();
+                if (list.Count == 0) engine.Snapshot.Storages.Delete(key);
+
+                //Mint GAS for oracle nodes
+                if (nodes.Length > 0)
+                {
+                    int index = (int)(response.Id % (ulong)nodes.Length);
+                    nodes[index].GAS += OracleRequestPrice;
+                }
+            }
+            foreach (var (account, gas) in nodes)
+            {
+                if (gas.Sign > 0) GAS.Mint(engine, account, gas);
+            }
+        }
+
+        [ContractMethod(OracleRequestPrice, CallFlags.AllowModifyStates)]
+        private void Request(ApplicationEngine engine, string url, string filter, string callback, StackItem userData, long gasForRepsonse)
+        {
+            //Check the arguments
+            if (Utility.StrictUTF8.GetByteCount(url) > MaxUrlLength
+                || (filter != null && Utility.StrictUTF8.GetByteCount(filter) > MaxFilterLength)
+                || Utility.StrictUTF8.GetByteCount(callback) > MaxCallbackLength
+                || gasForRepsonse <= 0)
+                throw new ArgumentException();
+
+            //Mint gas for the response
+            engine.AddGas(gasForRepsonse);
+            GAS.Mint(engine, Hash, gasForRepsonse);
```

Comment:
@erikzhang How about adding a `responseBasicFee`, and rename `gasForRepsonse` to `gasForCallback`?

```cs
engine.AddGas(responseBasicFee + gasForRepsonse);
GAS.Mint(engine, Hash, responseBasicFee + gasForRepsonse);
```

Reply:
Waht is the value of `responseBasicFee`?

Reply:
The basic network fee of response transaction.

Reply:
Can you calculate the exact value?

Reply:
The minimum size of reponse tx is 673,  therefore, the `responseBasicFee = 8675250` ( = tx.Size * GasPerByte + VerifyGas)


Here is the detail:
```cs
var script = new byte[0];
var sender = NativeContract.Oracle.Hash;
var cosigners = new Signer[]{
    new Signer
        {
            Account = Blockchain.GetConsensusAddress(oracleNodes),
            Scopes = WitnessScope.CalledByEntry,
            AllowedContracts = new UInt160[0],
            AllowedGroups = new ECPoint[0]
        }
};
var attributes = new TransactionAttribute[]
    {
        new OracleResponse
        {
            Id = 0,
            Success = false,
            Result = new byte[0]
        }
    };
ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, null, snapshot, 10 * (long)GasToken.GAS.Factor);
var tx = wallet.MakeTransaction(script, sender, cosigners, attributes);
// Sign code .....
System.Console.WriteLine(tx.Size);
System.Console.WriteLine(tx.NetworkFee);
```

Reply:
So 0.1 GAS is good, right?

Reply:
Yes.

Reply:
Done.